### ISSUE TITLE
New Game+ and New Game+2 Goal settings.

### ISSUE THREAD

**Issue by AuraOfTheDawn** [ThePhar/RogueLegacyRandomizer#71]
Archipelago multiworlds, especially large asyncs, tend to suffer from rogue legacy, given that every check is effectively a sphere 0 check. Inflating the count helps to give players something to do for a while, but after you get just a handful of decent items, the entire game can be stomped, and then you're done. Setting ng+ or ng+2 at the start of the game delays this by making everything early borderline inaccessible, but still turns into the same steamroll and quick exit, after a relatively trivial amount of effort.

I would like to see a 'goal' setting with options both to beat the fountain (as it is now), and to beat the fountain on ng+1/+2. (or random choice, because why not). This would appropriately lengthen your seed to help keep pace with other players.
As a side effect of this, you could divide the total number of chests placed in ng/ng+/ng+2 seperately, so that you'd create at least 3 spheres to tuck items in, rather then every item remaining a sphere 0. Also helps to reduce the need for a given player to 'grind' an area for checks in case something in there's helpful to someone else. Whether this should be X checks per (based off our existing sliders for customization, example: 30 checks per area per difficulty) or be X/Y/Z checks each (example: 50 checks total on a slider, dividing 20/10/20 or w/e other distribution a player might want) between ng/ng+1/ng+2 in a given area.

An alternative could be 'defeat the fountain X times' which might play nicer with people who like starting on higher ng+ levels, without shortening their goal. (a side effect is that your total run to 'defeat fountain at ng+2' might be over much quicker if you started on ng+2). You might also have to sanity check the goal versus where one starts. (make sure defeating the fountain at ng+2 when you start there qualifies for the goal of 'defeat fountain at ng+1', in case people randomize settings for goal and/or start mode)

Another alternative option that could slot neatly into rogue legacy and make use of this would be a literal check that shifts our games into ng+ or ng+2 regardless of our fountain kill status. This would work especially well as a sort of trap check to seed in a chest/other worlds, again making our progress potentially dependent more on other people, as we'd be unable to access checks in ng+ or ng+2 until our newgame+ upgrade was found in another world, making the experience tie in more with other players instead of being largely independent.

A knock-on effect of this when dividing chests between difficulty tiers is that you'd provide more meaning to our own progress in rogue legacy, whereas normally our progress is entirely and wildly independent from anyone elses.

**Comment by jfinalfunk** [comment]
I agree with the assertion that normal game is too easy, and NG+ is hard to start. I like the idea of starting with normal, beating fountain to progress to NG+, then beating that to finish! (and/or progress to NG++). Suggestion: This could also pair with fountain shards, which could be divided between Normal. Plus. and PlusPlus (or progressive?) shards, so add even more logical progression / gating to game completion, so that the experience extends from the start of an AP to the end. (Or at least to the point that all your shards drop). Great suggestion!
